src/dictionary.c

```diff
@@ -399,7 +484,11 @@ SEXP vctrs_id(SEXP x) {
 }
 
 // [[ register() ]]
-SEXP vec_match(SEXP needles, SEXP haystack) {
+SEXP vctrs_match(SEXP needles, SEXP haystack, SEXP na_equal) {
+  return vec_match_params(needles, haystack, Rf_asLogical(na_equal));
```

Comment:
Should we check `na_equal` more aggressively? Type + size + Not `NA` checks? (i.e. what `is_bool()` would do)

Reply:
Good point.